documentation/CAMARA-Security-Interoperability.md

```diff
@@ -0,0 +1,231 @@
+
+## Access Token Request
+
+This CAMARA document makes the scope parameter REQUIRED for the OAuth2 Client Credentials Grant.
+
+The client MUST authenticate with the authorization server as described in [Client Authentication Section](#client-authentication).
+
+## The Scope Parameter
+
+
+Scope values determine the specific CAMARA services being requested by the Service Provider, subject to the SP being registered to use those services. The scope values must be documented in the API OAS files by all Camara API subprojects. This document does not change OIDC definitions of scope values.
+
+
+---
+**NOTE**
+
+Scope values are an integral part of any OAuth2 and OIDC implementation. The RS enforces API access based on scope (if the Camara API subproject defines scopes).
+Therefore scopes should be available to API implementations.
+
+---
+
+
+## Missing "openid" scope
+
+[OIDC Core Authentication Request](https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest) states the following about the value of scope.
+> REQUIRED. OpenID Connect requests MUST contain the openid scope value. If the openid scope value is not present, the behavior is entirely unspecified. Other scope values MAY be present. ...
+
+This document defines the following error handling for a missing "openid" value in scope. 
+Please refer to [Authentication Error Response](https://openid.net/specs/openid-connect-core-1_0.html#AuthError).
+
+If "openid" is missing in the scope value, the Authorization Server returns an HTTP response code of 400 (Bad Request) and an error invalid_request.
+
+## Purpose
+
+A transaction specific request parameter purpose as specified in [openid-connect-4-identity-assurance-1_0-13](https://openid.net/specs/openid-connect-4-identity-assurance-1_0.html#name-transaction-specific-purpos) MUST be used to allow a SP to state the purpose for the transfer of End-User data it is asking for.
+The purpose string MUST use below format for interoperability
+
+`dpv:<dpvValue>` 
+
+`<dpvValue>` is coming from [W3C DPV purpose definition](https://w3c.github.io/dpv/dpv/#vocab-purpose)
+
+## ID Token sub claim
+
+To prevent [correlation](https://openid.net/specs/openid-connect-core-1_0.html#Correlation) OIDC recommends that the sub claim is a [Pairwise Pseudonymous Identifier (PPID)](https://openid.net/specs/openid-connect-core-1_0.html#Terminology).
+
+This document defines that the sub claim MUST be a Pairwise Pseudonymous Identifier (PPID).
+
+The PPID MUST not contain Personally Identifiable Information (PII) such as the MSISDN.
+If the PPID is based on PII then it must be hard for an attacker to reverse the PPID and reveal the PII.
+
+OIDC discusses [Pairwise Identifier Algorithms](https://openid.net/specs/openid-connect-core-1_0.html#PairwiseAlg).
+
+This document does not mandate a particular PPID algorithm to be used.
+
+
+## Client Authentication
+
+This CAMARA document allows **one** client authentication method, private_key_jwt, as defined in OIDC
```

Comment:
SimSwap deployment on prod faced _Client authentication challenge_ due to a lack of a common approach to request handling and client authentication solution.

If we mandate here _private_key_jwt_ then let's use it across CAMARA API in sub-projects that requires out-of-the band end-user authentication & consent acquisition. Additionally, we should have an agreement on _request object claims_ definition, as as described [in CIBA spec - signed auth request](https://openid.net/specs/openid-client-initiated-backchannel-authentication-core-1_0.html#signed_auth_request)

Reply:
The CAMARA OIDC profile, which is currently being defined, is precisely intended to solve these implementation problems and interoperability issues. Profiling specific CAMARA behaviors and definitions we all eventually agree.

Reply:
Improving interoperability and security regarding user authentication and user consent is exactly the purpose of the icm work.